ITISFoundation/osparc-simcore
# ♻️ Clean up simcore webserver environment

## What do these changes do?
Make env variables of Webserver clean, transparent and easy to understand. The `env_file` has been removed from docker-compoes `webesrver` service. The environment of docker-compose `webserver` service has been updated.

## Related issue/s

- closes https://github.com/ITISFoundation/osparc-ops-environments/issues/268
- related to https://github.com/ITISFoundation/osparc-simcore/issues/3635
- goes together with https://github.com/ITISFoundation/osparc-ops-environments/pull/273

## How to test
Deploy the whole stack and check whether simcore webserver is up and running without any errors in logs

## DevOps Checklist
Make sure to provide Webserver the following environment variables
Here are the environment variable names extracted from the given Docker Compose service:

- SWARM_STACK_NAME
- AIODEBUG_SLOW_DURATION_SECS
- DIAGNOSTICS_SLOW_DURATION_SECS
- STORAGE_PORT
- RESOURCE_USAGE_TRACKER_HOST
- WEBSERVER_DEV_FEATURES_ENABLED
- WEBSERVER_LOGLEVEL
- LOG_FORMAT_LOCAL_DEV_ENABLED
- WEBSERVER_HOST
- WEBSERVER_FRONTEND
- WEBSERVER_ACTIVITY
- WEBSERVER_PROMETHEUS_API_VERSION
- WEBSERVER_PROMETHEUS_URL
- WEBSERVER_CATALOG
- CATALOG_HOST
- CATALOG_PORT
- POSTGRES_DB
- POSTGRES_ENDPOINT
- POSTGRES_HOST
- POSTGRES_PASSWORD
- POSTGRES_PORT
- POSTGRES_USER
- WEBSERVER_DIAGNOSTICS
- DIAGNOSTICS_MAX_AVG_LATENCY
- DIAGNOSTICS_MAX_TASK_DELAY
- DIRECTOR_V2_HOST
- DIRECTOR_V2_PORT
- DIRECTOR_HOST
- DIRECTOR_PORT
- WEBSERVER_EMAIL
- SMTP_HOST
- SMTP_PORT
- WEBSERVER_EXPORTER
- WEBSERVER_GARBAGE_COLLECTOR
- INVITATIONS_HOST
- INVITATIONS_LOGLEVEL
- INVITATIONS_OSPARC_URL
- INVITATIONS_PASSWORD
- INVITATIONS_PORT
- INVITATIONS_SECRET_KEY
- INVITATIONS_USERNAME
- WEBSERVER_LOGIN
- LOGIN_REGISTRATION_CONFIRMATION_REQUIRED
- LOGIN_REGISTRATION_INVITATION_REQUIRED
- LOGIN_2FA_REQUIRED
- REDIS_HOST
- REDIS_PORT
- RESOURCE_MANAGER_RESOURCE_TTL_S
- WEBSERVER_SCICRUNCH
- SCICRUNCH_API_BASE_URL
- SCICRUNCH_API_KEY
- WEBSERVER_SESSION_SECRET_KEY
- WEBSERVER_STATICWEB
- STORAGE_ENDPOINT
- STORAGE_HOST
- STORAGE_PORT
- WEBSERVER_STUDIES_DISPATCHER
- STUDIES_ACCESS_ANONYMOUS_ALLOWED
- STUDIES_DEFAULT_SERVICE_THUMBNAIL
- WEBSERVER_TRACING
- TRACING_ENABLED
- TRACING_ZIPKIN_ENDPOINT
- TRACING_THRIFT_COMPACT_ENDPOINT
- WEBSERVER_PROJECTS
- PROJECTS_MAX_COPY_SIZE_BYTES
- PROJECTS_MAX_NUM_RUNNING_DYNAMIC_NODES
- RABBIT_HOST
- RABBIT_PASSWORD
- RABBIT_PORT
- RABBIT_USER
- WEBSERVER_GUNICORN_CMD_ARGS
- WEBSERVER_DB_LISTENER
- WEBSERVER_ANNOUNCEMENTS
- WEBSERVER_NOTIFICATIONS
- WEBSERVER_CLUSTERS
- WEBSERVER_GROUPS
- WEBSERVER_META_MODELING
- WEBSERVER_PRODUCTS
- WEBSERVER_PUBLICATIONS
- WEBSERVER_SOCKETIO
- WEBSERVER_TAGS
- WEBSERVER_USERS
- WEBSERVER_VERSION_CONTROL
- WB_GC_GARBAGE_COLLECTOR
- WB_GC_LOGLEVEL
- WB_GC_RESOURCE_MANAGER_RESOURCE_TTL_S
- WB_GC_ANNOUNCEMENTS
- WB_GC_ACTIVITY
- WB_GC_CATALOG
- WB_GC_NOTIFICATIONS
- WB_GC_DB_LISTENER
- WB_GC_DIAGNOSTICS
- WB_GC_EMAIL
- WB_GC_EXPORTER
- WB_GC_FRONTEND
- WB_GC_LOGIN
- WB_GC_PROJECTS
- WB_GC_SCICRUNCH
- WB_GC_STATICWEB
- WB_GC_TRACING
- WB_GC_CLUSTERS
- WB_GC_GROUPS
- WB_GC_META_MODELING
- WB_GC_PRODUCTS
- WB_GC_PUBLICATIONS
- WB_GC_SOCKETIO
- WB_GC_STUDIES_DISPATCHER
- WB_GC_TAGS
- WB_GC_USERS
- WB_GC_VERSION_CONTROL
- WB_DB_EL_LOGLEVEL
- WB_DB_EL_DB_LISTENER
- WB_DB_EL_SOCKETIO
- WB_DB_EL_ANNOUNCEMENTS
- WB_DB_EL_ACTIVITY
- WB_DB_EL_CATALOG
- WB_DB_EL_NOTIFICATIONS
- WB_DB_EL_DIAGNOSTICS
- WB_DB_EL_EMAIL
- WB_DB_EL_EXPORTER
- WB_DB_EL_FRONTEND
- WB_DB_EL_GARBAGE_COLLECTOR
- WB_DB_EL_LOGIN
- WB_DB_EL_PROJECTS
- WB_DB_EL_SCICRUNCH
- WB_DB_EL_STATICWEB
- WB_DB_EL_STORAGE
- WB_DB_EL_STUDIES_DISPATCHER
- WB_DB_EL_TRACING
- WB_DB_EL_CLUSTERS
- WB_DB_EL_GROUPS
- WB_DB_EL_META_MODELING
- WB_DB_EL_PRODUCTS
- WB_DB_EL_PUBLICATIONS
- WB_DB_EL_TAGS
- WB_DB_EL_USERS
- WB_DB_EL_VERSION_CONTROL
